src/Extensions/DBHTMLFieldExtension.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+
+namespace SilverStripe\SearchService\Extensions;
+
+use SilverStripe\Core\Extension;
+
+class DBHTMLFieldExtension extends Extension
+{
+    /**
+     * For HTML fields, we have to call ->forTemplate() so that shortcode get processed
+     *
+     * @param bool $shouldIncludeHTML
+     * @return string|string[]|null
+     */
+    public function getSearchValue(bool $shouldIncludeHTML = true)
+    {
+        if ($shouldIncludeHTML) {
+            return $this->owner->forTemplate();
+        }
+        return preg_replace('/\s+/S', " ", strip_tags($this->owner->forTemplate()));
```

Comment:
Let's just defer to the configuration here, rather than adding a new param. The param is only used in this one case, so I don't see a need for it (it's ignored in the generic implementation).